app/services/solidus_bolt/transactions/authorize.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Transactions
+    class Authorize < SolidusBolt::Base
+      attr_reader :order, :create_bolt_account, :credit_card
+
+      def initialize(order:, create_bolt_account:, credit_card:)
+        @order = order
+        @create_bolt_account = create_bolt_account
+        @credit_card = credit_card
+        super
+      end
+
+      def call
+        authorize
+      end
+
+      private
+
+      def authorize
+        url = "#{api_base_url}/#{api_version}/merchant/transactions/authorize"
+        HTTParty.post(url, headers: headers, body: body.to_json)
+      end
+
+      def body
+        {
+          auto_capture: false,
+          cart: order.bolt_cart,
+          credit_card: credit_card,
+          division_id: '', # used to be defined, now needs to stay empty for call to work
+          source: 'direct_payments',
+          user_identifier: order.bolt_user_identifier,
+          user_identity: order.bolt_user_identity,
+          create_bolt_account: create_bolt_account
+        }
+      end
+
+      def headers
+        {
+          'Content-Type' => 'application/json',
+          'X-Nonce' => generate_nonce,
+          'X-Publishable-Key' => publishable_key
+        }.merge!(authentication_header)
```

Comment:
why don't we use `authenticate(options)` instead, since it's already defined for this purpose?

Reply:
Because I found it hard to understand. 
It received just an argument to change the header array. 
I had to go into its definition to understand what was its purpose.